logging_handler.go

```diff
@@ -13,6 +13,10 @@ import (
 	"github.com/oauth2-proxy/oauth2-proxy/pkg/logger"
 )
 
+type GaggableResponseLogger interface {
```

Comment:
I'm not a fan of using the term `Gag` for this, is there another term we could use instead?

Reply:
Such as?

Reply:
How the other metadata is passed to `responseLogger` is in fake response headers, e.g. "GAP-Upstream-Address" (see just below in  `Write()` -> `ExtractGAPMetadata()`). So the responseLogger could stay the same, with just one more fake metadata header, e.g. "GAP-Quiet". Or, you could convert them all to use setter methods, and call the interface e.g. `MetaResponseLogger`. (The property could be called "mute" or "quiet" or "silent" or "no_log".)

Reply:
I see the `GAP-Upstream-Address` and `GAP-Auth` fields sort of piggybacking in the headers. They aren't really documented though, so it was unclear whether that was intentional or some sort of legacy leftover... Either way, using the request's headers as a "blackboard" in this way means it might accidentally clobber legitimate headers should someone have had them set. It also seems conflated with `hmacauth` with `GAP-Signature` somehow? I can fold in a rework of the two logging piggyback items I suppose, but that's a bigger scope than originally intended with this PR.

Reply:
I removed the interface at any rate and went with `silent` instead.

Reply:
looks good to me